### DISCUSSION TITLE
Model-Viewer and Threejs Integrations on Animations

### DISCUSSION THREAD

**Discussion by shamsail** [google/model-viewer#2033] [Q&A]
Hi, I wanted to have a look and get to know that where are the integrations of model-viewer with threejs? Because I wanted to work on the animations stuff but could not found any way around of ThreeJs. 
I tried with this. Please have a look 

```
let viewer = new ModelViewerElement();
viewer = document.getElementById("idOfViewer");
const scene = viewer[Object.getOwnPropertySymbols(viewer)[14]];

const clip = THREE.AnimationClip.findByName(scene.animations, "Animation Name");

const mixer = new THREE.AnimationMixer(scene);
const animationAction = mixer.clipAction(clip);
animationAction.enabled = true;
animationAction.setEffectiveTimeScale(1);
animationAction.setEffectiveWeight(1);
animationAction.reset();
animationAction.play();
```

But, it's not working.

I know there is a `viewer["animationName]="Animation Name";` `viewer.play();` but I want to have/work on AnimationMixer. But the way I'm doing , it's not playing any animation.

I would love to work on this side because I think there is some connection that I'm missing.

**Comment by elalish** [comment]
I think you're on the right track for hacking your way down into the three.js layer, but you're on your own for debug, as it's outside of our API.

**Reply by shamsail** [reply]
Can you please let me know that which things are included in the API and what will be included? As, model is currently just exposing the materials. (v1.5.0)

**Reply by elalish** [reply]
Private is hidden (as best we can in JS) behind symbols. The public API is directly visible on the modelviewer element. As for what will be included, that depends largely on the contributions we get. Feel free to open a PR and we can discuss the finer details of our API there.

**Reply by elalish** [reply]
We have a public animation API here, I was assuming that was somehow insufficient for you? https://modelviewer.dev/docs/#animation-attributes

**Reply by shamsail** [reply]
yeah, i did , please review the pr so that i can start and elaborate
